### DISCUSSION TITLE
Troubleshooting Variable Focused Vibration Intensity

### DISCUSSION THREAD

**Discussion by Neurosmith24** [kriswilk/buzzah#16] [General]
Troubleshooting an issue with our Buzzah! gloves and throwing a hail mary in case others might have ideas. We've got working set of gloves. The issue is that one finger's tactor seems to buzz with a very focused strong vibration-  focused on the 5mm contact point on top of the LRA. This is what I assume is the goal and the ideal. The other seven buzz, but they're all more general, not as focused on that contact point. I've tried shimming the LRAs to raise the contact points on the others, but that doesn't feel like it has brought the others up to the same focused vibration as the best one. For reference, all housings have 3mm of foam padding that the LRA sits on, and foam padding surrounding the LRA as well. Attached photo shows a couple, one with the finger plate off to show the LRA in the housing better. 

Any ideas on why one would be so different? In a previous discussion here @PWPInnovator898 talked about wiring from the "5V" pad on the microcontroller board to the "V+" pad on the multiplexer and switching the multiplexer's  switch to the "V+" setting in order to drive all of its output ports with a 5V power level. But @kriswilk suggested that shouldn't be necessary. So, I haven't tried that yet. Could that be the difference?

Oh, I should mention that I've tried swapping the really good tactor to a different fingers, by disconnect the wires from the RJ45 connector and connecting elsewhere - still vibrates more focused and strongly.

Thanks for any help or insight!

![IMG_0501](https://github.com/kriswilk/buzzah/assets/5769522/28f0d3fd-0ea4-48b8-8614-fb47dca8076d)

**Comment by PWPInnovator898** [comment]
Hello Neurosmith.  
Why one tactor performs so much better vibrationally than the others seems odd.  
Have you made any modifications to the code? One way to check whether each channel is identically configured would be to read out the register settings for each driver and compare.  Are you able to do this?   I use the Mu code editor for this kind of debug.  [Kris](https://github.com/kriswilk) may have better insight.  By the way, your tactor assemblies look fantastic!  Also,   I am powering the haptic drivers from the USB 5V supply directly and have not seen what you are describing

**Reply by Neurosmith24** [reply]
Thanks! I did update the code to be the same settings as you suggested previously, but no modifications other than the basic parameters.

I have no idea what checking the register settings for each driver are or how to do that so any suggestions how would be great. 

Yeah, I'm pretty happy with how the tactor housings came out. Just wish I could get all of them focused like the one. Thinking I'll try wiring the 5V supply directly and see what happens.

Thanks again!

**Reply by Neurosmith24** [reply]
Hi @PWPInnovator898 and @kriswilk I'm wondering if I may have missed something in using the VLV101040A LRAs. In the code, I specified the voltage and frequency to match those of that LRA. But the Vybronics page for that LRA says, " If using such a driver, you will need to disable the auto-resonance  calibration feature..." Was there some where or some way I'm supposed to explicitly disable auto-resonance for these LRAs? Thanks

**Reply by kriswilk** [reply]
@Neurosmith24 The auto-resonance feature is part of the driver, not the LRA. Drivers with auto-resonance calibration try to detect and lock on to the ideal frequency of the LRA. However with some LRAs that have wide operating frequency ranges, the auto-resonance feature can get confused.

The current code drives the LRA manually, with the auto-resonance feature turned off. That's why the frequency needs to be specified in the header of the code.

**Comment by kriswilk** [comment]
Hello all,

I'm not sure why one LRA would feel so different. Have you tried switching the LRA to a different driver to make sure it's not being driven differently than the others?

Assuming that doesn't change anything, one possibility is that the natural frequency of that particular LRA is better matched to the specified driving frequency. You might try adjusting the software to drive the system at slightly different frequencies and observe if/how the intensities of the LRAs change.

**Reply by Neurosmith24** [reply]
Thanks @kriswilk ! I did switch the LRA to connect to other drivers and still got the same result. I just tried @PWPInnovator898's idea of wiring from the microcontroller 5V to mux's V1+ and switched the tiny switch. That feels like it made a significant difference in many, but not all of the LRAs. Now there's about 4 or 5 that match or closely match the 'good' one. A couple of others though still don't seem up to the others. 

The natural frequency of individual LRAs matching the specified driving frequency that you mentioned makes sense now that you mention it (I never would have thought of that). I like the idea of adjusting the frequencies in the code, so I'll try that next. I already have it set to 235 hz, so I assume I'd need to adjust it down since I believe that's the max for these LRAs.

Thanks again @kriswilk and @PWPInnovator898 for your time and input here. I really appreciate it!

**Reply by PWPInnovator898** [reply]
As [Kris ](https://github.com/kriswilk) indicated, there is a strong frequency dependent vibration intensity response for the LRAs, as noted in the datasheet for VLV1040A:  The LRAs will operate in the range noted on the x-axis, with a peak at 170 Hz.
![image](https://github.com/kriswilk/buzzah/assets/128418916/8fd279bb-0cf5-48ff-b48a-4e54c7ab736c).
In my layman's interpretation, this isn't necessarily "bad", as stimulation intensity is yet another randomization variable Tass and others mention in later "double randomization" papers. https://www.frontiersin.org/articles/10.3389/fnetp.2022.864859/full  
So, per an earlier request, Kris added the ability to randomly adjust stimulation intensity at a fixed frequency with the code variables AMPLITUDE_MIN and AMPLITUDE_MAX. I currently use 65 and 100 with ok results.
I also interpret the stimulation frequency (in the approximate range of 200 to 300 Hz) seeming to be less important than the VCR patterns themselves, as Dr. Tass vaguely noted in one of his webinars last year.
Also, just to confirm, I connect the USB 5V supply on the MCU module directly to the power supply pins of each TI haptic driver.
p.s. Would you mind sharing your latest 3D design file for your latest tactor holder?  Thank you.

**Reply by Neurosmith24** [reply]
Been meaning to follow up with an update. I lowered the frequency in the code down to 175 and that made an enormous difference in several of the LRAs. Now I have 4 that are very strong and focused and four that are less so, but better than they were. Perhaps coincidentally, we noticed some improvements this past week that we've been running at these settings. 

I'm considering increasing the frequency a bit closer to 200 Hz and see how the LRAs respond at that level. Considering buying some more of these LRAs to see if I can get more strong ones. Thanks for you help and input guys!

**Reply by PWPInnovator898** [reply]
Thank you for the update.
And I’m very pleased that you’ve seen therapeutic benefits.
I’m working on a Bluetooth-controlled version of the buzzah electronics, and it’s coming along quite well with a very small PCB. It is not quite ready to share yet, but as soon as I have everything working, I’ll let you know.

**Reply by TactileDesign** [reply]
Hello PWPInnovator898, Thank you for all of your contributions here.  They have been very helpful as I have started putting my own build together.  Regarding your bluetooth version of the Buzzah, can you share any more details on that?   My understanding is that you implementing bluetooth in order to be able to adjust parameters on-the-fly, without having to stop and reprogram the Buzzah with a computer.  Is that correct?

**Reply by PWPInnovator898** [reply]
Yes. That is correct.   The Bluetooth version has been prototyped and works.   I’m now revising code to more accurately sync the L and R hands for users who think they need that feature.  Note - I’ve not read anything from Dr. Tass that directly states or implies sync across hands is required. What’s left for  me is to get a bug fixed in The File Glider application from Adafruit that enables code changes over Bluetooth without a WIRED USB connection.

**Reply by TactileDesign** [reply]
Thanks for your response. I will be eager to implement your new design when you get it working!  Regarding your syncing across hands comment - Like you, I have also not heard or seen anything from Dr. Tass explicitly saying that syncing is required for the process to work.  For example, there is a guy in Europe I saw on Youtube who developed very nice self contained systems for each glove that are not synchronized - so no wire or bluetooth connection between gloves.  Also, in response to a question in Youtube presentation (I think it was Nosilverbullet4PD), Dr. Tass compared his glove system to DBS and said that, since DBS works for the whole brain when only installed on one side, he would expect his glove to work even if only one hand was stimulated - although he has not tested this conjecture.

That said, I am leaning towards syncing for two reasons:  

1) In both of Dr. Tass’s patterns outlined in the Frontiers paper, he synchronizes the pulses between hands so that the rest period is coincident on both sides.  So, presumably the Tass group had the option to sync or not to sync, and they made the conscious decision to sync.  And all the encouraging data we have seen comes from patients who were stimulated with synced gloves.  

2) When I am “in the groove” with a session using my current ERM gloves, I can sense waves of relaxation sweep over me during the resting periods.  If gloves were running out of lock, they would be syncing up sometimes only to phase out of phase after a while since the periods would not be phase-locked - creating a low beat frequency that I might become aware of.  So I think that scenario would be annoying instead of relaxing, and it seems to me that the relaxation period is important to the process.

I hope to have my Buzzah system up and running for testing today!
